### PR TITLE
[1LP][RFR] This PR will automate SSA via Compliance.

### PR DESCRIPTION
**Pull request by sbulage**
PRT run:
{{ pytest: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py -k test_ssa_compliance -vvvv --use-provider vsphere6-nested --long-running }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9d512f44225ac9730000848b180f3950e16d21e2:

`cfme/tests/cloud_infra_common/test_vm_instance_analysis.py`:
- :bangbang: Line 203:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 205:13: **E121** *continuation line under-indented for hanging indent*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py [hunk 1]
```diff
@@ -371,6 +404,83 @@ def test_ssa_template(local_setup_provider, provider, soft_assert, vm_analysis_p
 
 
 @pytest.mark.rhv3
+@pytest.mark.tier(2)
+@pytest.mark.long_running
+def test_ssa_compliance(local_setup_provider, ssa_compliance_profile, ssa_profiled_vm, ssa_vm,
```
Comment (rlbabyuk) [inline cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: Question: Don't understand why we need ssa_profiled_vm and ssa_vm they are the same

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py [hunk 2]
```diff
@@ -371,6 +404,83 @@ def test_ssa_template(local_setup_provider, provider, soft_assert, vm_analysis_p
...
+        e_groups = ssa_vm.ssh.run_command("cat /etc/group | wc -l").output.strip('\n')
+        e_packages = ssa_vm.ssh.run_command(
+            ssa_vm.system_type['package-number']).output.strip('\n')
+        e_services = ssa_vm.ssh.run_command(
+            ssa_vm.system_type['services-number']).output.strip('\n')
+
+    logger.info("Expecting to have %s users, %s groups, %s packages and %s services", e_users,
+                e_groups, e_packages, e_services)
+    view = appliance.browser.create_view(TasksView)
+    wait_for(
```
Comment (rlbabyuk) [inline cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: Required: Please add message for wait for 
ex. message="Waiting for SSA runs for {} vm".format(ssa_vm.name)

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py [hunk 3]
```diff
@@ -163,18 +163,50 @@ def local_setup_provider(request, setup_provider_modscope, provider, appliance):
...
+    yield
+    provider.unassign_policy_profiles(profile.description)
+    profile.delete()
+
+
 @pytest.fixture(scope="module")
 def ssa_vm(request, local_setup_provider, provider, vm_analysis_provisioning_data,
            appliance, analysis_type):
     """ Fixture to provision instance on the provider """
+    template_name = template_name = vm_analysis_provisioning_data['image']
```
[the remark this answers is not in the document]
Comment (sbulage) [reply cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: oops typo :)

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py (hunk 2 above)
Comment (sbulage) [reply cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: Okay will add it.

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py [hunk 4]
```diff
@@ -163,18 +163,50 @@ def local_setup_provider(request, setup_provider_modscope, provider, appliance):
...
+@pytest.fixture(scope="module")
+def ssa_compliance_policy(appliance):
+    policy = appliance.collections.policies.create(
+        VMControlPolicy,
+        'ssa_policy_{}'.format(fauxfactory.gen_alpha())
+    )
+    policy.assign_events("VM Provision Complete")
+    policy.assign_actions_to_event("VM Provision Complete", ["Initiate SmartState Analysis for VM"])
+    yield policy
+    policy.assign_events()
```
[the remark this answers is not in the document]
Comment (sbulage) [reply cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: Adding policy.delete()

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py [hunk 5]
```diff
@@ -264,12 +297,12 @@ def ssa_policy(appliance, ssa_action):
 
 
 @pytest.fixture(scope="module")
-def ssa_profile(appliance, ssa_vm, ssa_policy):
+def ssa_profiled_vm(appliance, ssa_vm, ssa_policy):
```
[the remark this answers is not in the document]
Comment (sbulage) [reply cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: yes @rlbabyuk you are right, but after assigning profile to VM it is returning ssa_profile which is not used in the rest of the test. So, I have directly returned ssa_vm from this fixture only.

[the remark this answers is not in the document]
Comment (rlbabyuk) [reply cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: @sbulage in that case you don't need to use ssa_vm in other tests, this is a fixture duplication, just replace ssa_vm to ssa_profiled_vm for tests

[the remark this answers is not in the document]
Comment (rlbabyuk) [reply cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: @sbulage sorry, I've done this already ;)

[the remark this answers is not in the document]
Comment (sbulage) [reply cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: @rlbabyuk Already, I have removed it from the tests :)

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py [hunk 6]
```diff
@@ -287,7 +321,7 @@ def detect_system_type(vm):
 
 
 @pytest.fixture(scope="module")
-def schedule_ssa(appliance, ssa_vm, ssa_profile, wait_for_task_result=True):
+def schedule_ssa(appliance, ssa_vm, ssa_profiled_vm, wait_for_task_result=True):
```
Comment (rlbabyuk) [inline cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: Required: please remove ssa_vm in the tests when you use ssa_profiled_vm and update ssa_vm in the tests respectively

Comment (sbulage) [reply cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: @rlbabyuk I am working on it and will raise separate PR with all code refactored.

Comment (sbulage) [reply cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: Yes @rlbabyuk you are right, locally I have tested it.

**Pull request merged**